core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/FileScanSourceOpExec.scala

```diff
@@ -0,0 +1,68 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeTypeUtils
+import org.apache.commons.io.IOUtils.toByteArray
+
+import java.io._
+import java.nio.file.{Files, Paths}
+import java.util.zip.ZipFile
+import scala.jdk.CollectionConverters.{
+  asScalaIteratorConverter,
+  enumerationAsScalaIteratorConverter
+}
+
+class FileScanSourceOpExec private[text] (val desc: FileScanSourceOpDesc)
+    extends SourceOperatorExecutor {
+
+  @throws[IOException]
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    if (desc.unzip) {
+      val zipReader = new ZipFile(desc.filePath.get)
+      val entries =
+        zipReader.entries().asScala.filterNot(entry => entry.getName.startsWith("__MACOSX"))
+      if (desc.attributeType.isSingle) {
+        entries.map(entry => singleTuple(toByteArray(zipReader.getInputStream(entry))))
+      } else {
+        entries.flatMap(entry => multipleTuple(zipReader.getInputStream(entry)))
+      }
+    } else {
+      if (desc.attributeType.isSingle) {
+        Iterator(singleTuple(Files.readAllBytes(Paths.get(desc.filePath.get))))
+      } else {
+        multipleTuple(new FileInputStream(desc.filePath.get))
+      }
+    }
+  }
+
+  private def singleTuple(file: Array[Byte]): Tuple =
+    new Tuple(
+      desc.sourceSchema(),
+      desc.attributeType match {
+        case FileAttributeType.BINARY        => file
+        case FileAttributeType.SINGLE_STRING => new String(file, desc.encoding.getCharset)
+      }
+    )
+
+  private def multipleTuple(stream: InputStream): Iterator[Tuple] = {
+    new BufferedReader(new InputStreamReader(stream, desc.encoding.getCharset))
+      .lines()
+      .iterator()
+      .asScala
+      .slice(
+        desc.fileScanOffset.getOrElse(0),
+        desc.fileScanOffset.getOrElse(0) + desc.fileScanLimit.getOrElse(Int.MaxValue)
+      )
+      .map(line =>
+        new Tuple(
+          desc.sourceSchema(),
+          AttributeTypeUtils.parseField(line, desc.attributeType.getType)
+        )
+      )
+  }
```

Comment:
some minor comments regarding style:
1. `singleTuple` and `multipleTuple` as function names are confusing. Try to use verbs in the function name to describe what does this function do. Also, it should at least be `multipleTuples` (plural form).
2. They are parallel functions but their signature are different. it is confusing to use. Try to align their input parameters? Also maybe the multi-tuple function can call single tuple internally to avoid duplicate code.

Reply:
Fixed by @Yicong-Huang.